Automattic/jetpack
# Podcast Player Block: Add basic playlist and player UI

Update: Cherry picked commits from https://github.com/Automattic/jetpack/pull/14952 in order to make this easier to merge into master.

updated by @marekhrabe:

#### Changes proposed in this Pull Request:

* This PR adds a basic HTML output with Podcast episodes listed. 
* On the frontend, an audio player will show up above the list of episodes.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
paYJgx-sH-p2

#### Testing instructions:

- rebuild extensions
- insert Podcast Player block in the editor and give it a podcast feed (for example https://distributed.blog/category/podcast/feed/)
- wait for the content to load and you should see a list of episodes inside a box with an additional border
- change number of episodes to show in the sidebar
- content should update
- save and open on the frontend (possible by clicking the Preview button)
- you should see an audio player UI
- clicking any episode link stops the current one and starts loading your selected one
- you should hear audio and be able to control it using the player UI

There are many aspects of the player that are not finished. This PR will only include what's described above and the rest will be implemented in subsequent PRs.

Expected look in the editor:

<img width="644" alt="Screenshot 2020-03-13 at 14 21 03" src="https://user-images.githubusercontent.com/156676/76625410-73e1b580-6537-11ea-8e8b-7fb65198d8d0.png">

Expected look on the frontend:

<img width="611" alt="Screenshot 2020-03-13 at 14 23 19" src="https://user-images.githubusercontent.com/156676/76625424-7b08c380-6537-11ea-9efb-72e7aa244d84.png">


#### Proposed changelog entry for your changes:

* none (work in progress)

Fixes #14941.
Fixes #14942.
